test/org/sagebionetworks/bridge/models/upload/UploadSchemaTest.java

```diff
@@ -259,58 +267,61 @@ public void testSerialization() throws Exception {
-        assertEquals(13, jsonMap.size());
-        assertEquals("Test Schema", jsonMap.get("name"));
-        assertEquals(3, jsonMap.get("revision"));
-        assertEquals("test-schema", jsonMap.get("schemaId"));
-        assertEquals("ios_survey", jsonMap.get("schemaType"));
-        assertEquals("test-study", jsonMap.get("studyId"));
-        assertEquals("survey-guid", jsonMap.get("surveyGuid"));
-        assertEquals("UploadSchema", jsonMap.get("type"));
-        assertEquals(6,  jsonMap.get("version"));
-        assertEquals(true, jsonMap.get("published"));
-
-        Map<String, Integer> maxAppVersionMap = (Map<String, Integer>) jsonMap.get("maxAppVersions");
+        JsonNode jsonNode = BridgeObjectMapper.get().readTree(convertedJson);
+        assertEquals(15, jsonNode.size());
+        assertEquals(MODULE_ID, jsonNode.get("moduleId").textValue());
+        assertEquals(MODULE_VERSION, jsonNode.get("moduleVersion").intValue());
+        assertEquals("Test Schema", jsonNode.get("name").textValue());
+        assertEquals(3, jsonNode.get("revision").intValue());
+        assertEquals("test-schema", jsonNode.get("schemaId").textValue());
+        assertEquals("ios_survey", jsonNode.get("schemaType").textValue());
+        assertEquals("test-study", jsonNode.get("studyId").textValue());
+        assertEquals("survey-guid", jsonNode.get("surveyGuid").textValue());
+        assertEquals("UploadSchema", jsonNode.get("type").textValue());
+        assertEquals(6,  jsonNode.get("version").intValue());
+        assertTrue(jsonNode.get("published").booleanValue());
+
+        JsonNode maxAppVersionMap = jsonNode.get("maxAppVersions");
         assertEquals(2, maxAppVersionMap.size());
         assertEquals(37, maxAppVersionMap.get("iOS").intValue());
         assertEquals(42, maxAppVersionMap.get("Android").intValue());
 
-        Map<String, Integer> minAppVersionMap = (Map<String, Integer>) jsonMap.get("minAppVersions");
+        JsonNode minAppVersionMap = jsonNode.get("minAppVersions");
         assertEquals(2, minAppVersionMap.size());
         assertEquals(13, minAppVersionMap.get("iOS").intValue());
         assertEquals(23, minAppVersionMap.get("Android").intValue());
 
         // The createdOn time is converted into ISO timestamp, but might be in a different timezone. Ensure that it
         // still refers to the correct instant in time, down to the millisecond.
-        long resultSurveyCreatedOnMillis = DateTime.parse((String) jsonMap.get("surveyCreatedOn")).getMillis();
+        long resultSurveyCreatedOnMillis = DateTime.parse(jsonNode.get("surveyCreatedOn").textValue()).getMillis();
         assertEquals(surveyCreatedOnMillis, resultSurveyCreatedOnMillis);
 
-        List<Map<String, Object>> fieldDefJsonList = (List<Map<String, Object>>) jsonMap.get("fieldDefinitions");
+        JsonNode fieldDefJsonList = jsonNode.get("fieldDefinitions");
```

Comment:
The Map seemed kind of clunky. If we wanted to test JSON, it was better to test JSON directly instead of translating it into a Map.